core/controllers/cron_test.py

```diff
@@ -316,6 +321,149 @@ def _mock_logging_function(msg, *args):
         )
 
 
+class CronMailContributorDashboardReviewerOpportunitiesHandlerTests(
+        test_utils.GenericTestBase):
+
+    target_id = 'exp1'
+    language_code = 'en'
+    default_translation_html = '<p>Sample translation</p>'
+    AUTHOR_USERNAME = 'author'
+    AUTHOR_EMAIL = 'author@example.com'
+    REVIEWER_USERNAME = 'reviewer'
+    REVIEWER_EMAIL = 'reviewer@community.org'
+
+    def _create_translation_suggestion(self):
+        """Creates a translation suggestion."""
+        add_translation_change_dict = {
+            'cmd': exp_domain.CMD_ADD_TRANSLATION,
+            'state_name': feconf.DEFAULT_INIT_STATE_NAME,
+            'content_id': feconf.DEFAULT_NEW_STATE_CONTENT_ID,
+            'language_code': self.language_code,
+            'content_html': feconf.DEFAULT_INIT_STATE_CONTENT_STR,
+            'translation_html': self.default_translation_html
+        }
+
+        return suggestion_services.create_suggestion(
+            suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+            suggestion_models.TARGET_TYPE_EXPLORATION,
+            self.target_id, feconf.CURRENT_STATE_SCHEMA_VERSION,
+            self.author_id, add_translation_change_dict,
+            'test description')
+
+    def _assert_reviewable_suggestion_email_infos_are_equal(
+            self, reviewable_suggestion_email_info,
+            expected_reviewable_suggestion_email_info):
+        """Asserts that the reviewable suggestion email info is equal to the
+        expected reviewable suggestion email info.
+        """
+        self.assertEqual(
+            reviewable_suggestion_email_info.suggestion_type,
+            expected_reviewable_suggestion_email_info.suggestion_type)
+        self.assertEqual(
+            reviewable_suggestion_email_info.language_code,
+            expected_reviewable_suggestion_email_info.language_code)
+        self.assertEqual(
+            reviewable_suggestion_email_info.suggestion_content,
+            expected_reviewable_suggestion_email_info.suggestion_content)
+        self.assertEqual(
+            reviewable_suggestion_email_info.submission_datetime,
+            expected_reviewable_suggestion_email_info.submission_datetime)
+
+    def _mock_send_contributor_dashboard_reviewers_emails(
+            self, reviewer_ids, reviewers_suggestion_email_infos):
+        """Mocks
+        email_manager.send_mail_to_notify_contributor_dashboard_reviewers as
+        its not possible to send mail with self.testapp_swap, i.e with the URLs
+        defined in main_cron.
+        """
+        self.reviewer_ids = reviewer_ids
+        self.reviewers_suggestion_email_infos = reviewers_suggestion_email_infos
```

Comment:
Don't we need to reset this state after the mock method is no longer being used?